src/coord/src/catalog/migrate.rs

```diff
@@ -190,30 +190,34 @@ fn ast_rewrite_index_sink_cluster_default_0_23_0(
     stmt: &mut mz_sql::ast::Statement<Raw>,
 ) -> Result<(), anyhow::Error> {
     match stmt {
-        Statement::CreateIndex(CreateIndexStatement {
-            name: _,
-            in_cluster,
-            on_name: _,
-            key_parts: _,
-            with_options: _,
-            if_not_exists: _,
-        }) => {
-            *in_cluster = Some(RawIdent::Resolved(DEFAULT_COMPUTE_INSTANCE_ID.to_string()));
+        Statement::CreateIndex(
+            stmt @ CreateIndexStatement {
+                name: _,
+                in_cluster: None,
+                on_name: _,
+                key_parts: _,
+                with_options: _,
+                if_not_exists: _,
+            },
```

Comment:
Personally I'd probably skip the explicit field ignore and just write

```rust
Statement::CreateIndex(CreateIndexStatement { in_cluster: in_cluster @ None }) => {
    *in_cluster = Some(...);
}
```

on the theory that this migration is concerned with precisely the one field that it's already looking at it, and adding a new field to the `CreateIndexStatement` doesn't need to require that you come understand this migration and see if the field needs special handling.